src/main/drivers/accgyro/accgyro_spi_icm456xx.c

```diff
@@ -490,16 +521,78 @@ bool icm456xxSpiAccDetect(accDev_t *acc)
 
 bool icm456xxGyroReadSPI(gyroDev_t *gyro)
 {
-    uint8_t raw[6] = {0};
+    int16_t *gyroData = (int16_t *)gyro->dev.rxBuf;
+    switch (gyro->gyroModeSPI) {
+    case GYRO_EXTI_INIT:
+    {
+        // Initialise the tx buffer to all 0xff
+        memset(gyro->dev.txBuf, 0xff, 12);
+
+        // Check that minimum number of interrupts have been detected
+
+        // We need some offset from the gyro interrupts to ensure sampling after the interrupt
+        gyro->gyroDmaMaxDuration = 5;
+        if (gyro->detectedEXTI > GYRO_EXTI_DETECT_THRESHOLD) {
+#ifdef USE_DMA
+            if (spiUseDMA(&gyro->dev)) {
+                gyro->dev.callbackArg = (uint32_t)gyro;
+                gyro->dev.txBuf[0] = ICM456XX_GYRO_DATA_X1_UI | 0x80;
+                gyro->segments[0].len = 12;
```

Comment:
Why 12? command + 11 bytes, but only 6 are used?

Reply:
I don't fully understand the SPI DMA API. Need 6 byte

Reply:
You need to transfer register address to - that is 7 bytes ...